src/mlpack/core/dists/normal_distribution_impl.hpp

```diff
@@ -0,0 +1,53 @@
+/**
+ * @file normal_distribution.cpp
+ * @author Atharva Khandait
+ *
+ * Implementation of Normal distribution class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DISTRIBUTIONS_NORMAL_DISTRIBUTION_IMPL_HPP
+#define MLPACK_CORE_DISTRIBUTIONS_NORMAL_DISTRIBUTION_IMPL_HPP
+
+#include "normal_distribution.hpp"
+
+using namespace mlpack;
+using namespace mlpack::distribution;
+
+template<typename DataType>
+NormalDistribution<DataType>::NormalDistribution(
+    const DataType mean,
+    const DataType stdDeviation) :
+    mean(mean),
+    stdDeviation(stdDeviation)
+{
+  // Nothing to do here.
+}
+
+template<typename DataType>
+DataType NormalDistribution<DataType>::Random() const
+{
+  return stdDeviation % arma::randn<DataType>(mean.n_rows, mean.n_cols) + mean;
+}
+
+template<typename DataType>
+double NormalDistribution<DataType>::LogProbability(
+    const DataType& observation) const
+{
+  return -0.5 * (arma::accu(2 * arma::log(stdDeviation) +
+      arma::pow(std::move(mean - observation) / stdDeviation, 2) + log2pi));
```

Comment:
Is there a need for `std::move()` here?  `mean - observation` will already be a temporary, so I don't think it will make any difference (it might make things slower).